Lib/pdb.py

```diff
@@ -590,6 +591,36 @@ def default(self, line):
         except:
             self._error_exc()
 
+    def _replace_convenience_variable(self, line):
+        """Replace the convenience variable in line"""
+        last_token_is_dollar = False
```

Comment:
If the line doesn't have any `$` (most cases), there's no need to run the tokenizer at all.